src/formats/maeformat.cpp

```diff
@@ -0,0 +1,333 @@
+/**********************************************************************
+Copyright (C) 2019 by Pat Lorton
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+***********************************************************************/
+
+/* Populate an OBMol from a Schrödinger Maestro file using MaeParser.
+   More information can be found at: https://github.com/schrodinger/maeparser
+*/
+
+#include <openbabel/babelconfig.h>
+#include <openbabel/obmolecformat.h>
+
+#include <openbabel/obmolecformat.h>
+#include <openbabel/mol.h>
+#include <openbabel/atom.h>
+#include <openbabel/bond.h>
+#include <openbabel/obiter.h>
+#include <openbabel/elements.h>
+#include <openbabel/generic.h>
+
+#include <iostream>
+#include <map>
+
+#include <MaeConstants.hpp>
+#include <Reader.hpp>
+#include <Writer.hpp>
+
+using namespace std;
+using namespace schrodinger::mae;
+using boost::dynamic_bitset;
+namespace OpenBabel
+{
+
+class MAEFormat : public OBMoleculeFormat
+{
+public:
+  //Register this format type ID in the constructor
+  MAEFormat()
+	{
+		OBConversion::RegisterFormat("MAE", this);
+		OBConversion::RegisterFormat("MAEGZ", this);
```

Comment:
This will allow for both passing -imae or -imaegz format on the cmdline, or automatically if anything ends in .mae or .maegz it will use this registered extension.  What's *especially* nice (and undocumented as far as I can find), is anything ending in 'gz' it automatically gunzips on the input filestream, so we have to do nothing special to read maegz.